noxfile.py

```diff
@@ -180,3 +181,38 @@ def make(session: nox.Session):
         *(args.make_args or ("clean", "coredocs")),
     ]
     session.run("make", "-C", "docs/docsite", *make_args, external=True)
+
+
+def _rm_tmpdir(tmpdir):
+    rmtree(tmpdir, ignore_errors=True)
+
+
+@nox.session
+def tag(session: nox.Session):
+    """
+    Determine the missing ansible-core releases,
+    create corresponding tags for each release in the ansible-documentation repo,
+    and then push them
+    """
+    install(session, req="tag")
+
+    DEFAULT_REPO = "https://github.com/ansible/ansible"
+    CHECK_LOCATION = Path.cwd().parents[0] / "ansible"
+
+    args = list(session.posargs)
+    if not any(arg.startswith("--core") for arg in args):
+        if (CHECK_LOCATION / ".git").is_dir():
+            session.run("git", "-C", CHECK_LOCATION, "pull", "--quiet", external=True)
+            args.extend(["--core", str(CHECK_LOCATION)])
```

Comment:
We should probably have some try blocks in here to handle errors a little more gracefully. Something like:

```
try:
  session.run("git", "-C", CHECK_LOCATION, "pull", "--quiet", external=True)
  args.extend(["--core", str(CHECK_LOCATION)])
except Exception as e:
  session.error(f"Could not update the core repository: {e}")
```

However, in my case, I ran into `fatal: Need to specify how to reconcile divergent branches.` when trying this out. I think to overcome that we'd need to use a pull strategy in the nox session. I'm a `--rebase` kind of guy but other folks have different preferences so it's tricky to find any one right way to handle this.

In any event I guess the larger concern here might be that a pull could risk wiping local changes or creating a mess on someone's branch. It seems we're also going down the road of adding extending the tagger via the nox session, which is something I'd prefer not to do. Although I know it's sort of my fault for suggesting we compare the local checkout with the upstream core repo in an earlier comment.

Git operations can be finicky so maybe it's best to go with your initial approach of cloning into the virtual environment. If we can remove the clone when we're done that probably takes care of the concern about being a bit wasteful.